addons/pos_pine_labs/static/src/app/utils/payment/payment_pine_labs.js

```diff
@@ -0,0 +1,260 @@
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { serializeDateTime } from "@web/core/l10n/dates";
+import { offlineErrorHandler, handleRPCError } from "@point_of_sale/app/errors/error_handlers";
+import { register_payment_method } from "@point_of_sale/app/store/pos_store";
+import { ConnectionLostError, RPCError } from "@web/core/network/rpc";
+
+const REQUEST_TIMEOUT_MS = 5000;
+const CANCEL_REQUEST_TIME_LIMIT_MS = 125000;
+const { DateTime } = luxon;
+
+export class PaymentPineLabs extends PaymentInterface {
+    setup() {
+        super.setup(...arguments);
+        this.pollingTimeout = null;
+        this.inactivityTimeout = null;
+        this.payment_stopped = false;
+    }
+
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._processPineLabs();
+    }
+
+    pendingPineLabsPaymentLine() {
+        return this.pos.getPendingPaymentLine("pine_labs");
+    }
+
+    send_payment_cancel(order, cid) {
+        super.send_payment_cancel(order, cid);
+        return this._pineLabsCancel();
+    }
+
+    _callPineLabs(data, action) {
+        return this.pos.data
+            .call("pos.payment.method", action, [[this.payment_method_id.id], data])
+            .catch((error) => {
+                const line = this.pendingPineLabsPaymentLine();
+                this.pos.paymentTerminalInProgress = false;
+                if (line) {
+                    line.set_payment_status("force_done");
+                }
+                if (error instanceof ConnectionLostError) {
+                    offlineErrorHandler(this.env, error, error);
+                } else if (error instanceof RPCError) {
+                    handleRPCError(error, this.env.services.dialog);
+                } else {
+                    throw error;
+                }
+            });
+    }
+
+    /**
+     * Handles the response from Pine Labs for a make payment request.
+     * @param {Object} response - The response received from Pine Labs.
+     * @returns {Promise<boolean>} - Resolves when the payment confirmation process completes.
+     */
+    async _makePaymentRequestHandler(response) {
```

Comment:
The `@returns` docs isn't correct. The method doesn't only resolve to `Promise<boolean>`, it also resolves to the `pine_labs_fetch_payment_status` response when the transaction status is approved. Is that intended?